src/WidgetBase.ts

```diff
@@ -429,26 +452,46 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 		let nodes = Array.isArray(node) ? [ ...node ] : [ node ];
 		while (nodes.length) {
 			const node = nodes.pop();
-			if (isHNode(node)) {
-				if (node.properties.key) {
-					node.properties.afterCreate = this._afterCreateCallback;
-					node.properties.afterUpdate = this._afterUpdateCallback;
+			if (isHNode(node) || isWNode(node)) {
+				node.properties = node.properties || {};
+				if (isHNode(node)) {
+					if (node.properties.key) {
+						node.properties.afterCreate = this._afterCreateCallback;
+						node.properties.afterUpdate = this._afterUpdateCallback;
+					}
+					if (node.properties.bind === undefined) {
+						(<any> node.properties).bind = this;
+					}
 				}
+				else {
+					const { defaultRegistry, registry } = this.getCoreProperties(node.properties);
+					node.coreProperties = node.coreProperties || {};
+
+					if (node.coreProperties.bind === undefined) {
+						node.coreProperties.bind = this;
+					}
+					node.coreProperties.defaultRegistry = defaultRegistry;
+					node.coreProperties.registry = registry;
+				}
+				nodes = [ ...nodes, ...node.children ];
 			}
-			else if (!isWNode(node)) {
-				continue;
-			}
-			node.properties = node.properties || {};
-			if ((<any> node.properties).bind === undefined) {
-				(<any> node.properties).bind = this;
-			}
-			if (isWNode(node)) {
-				(<any> node.properties).defaultRegistry = this._registries.defaultRegistry;
-			}
-			nodes = [ ...nodes, ...node.children ];
 		}
 	}
 
+	protected getCoreProperties(properties: any): CoreProperties {
```

Comment:
do you think this should be double underscored?

Reply:
That's a reasonable suggestion as we do not want people to override this function, even if it is just `protected`